components/locid/src/helpers.rs

```diff
@@ -44,12 +45,84 @@ impl<T> ShortVec<T> {
         }
     }
 
+    #[inline]
+    pub fn as_mut_slice(&mut self) -> &mut [T] {
+        match self {
+            ShortVec::Empty => &mut [],
+            ShortVec::Single(v) => core::slice::from_mut(v),
+            ShortVec::Multi(v) => v.as_mut_slice(),
+        }
+    }
+
+    #[inline]
     pub const fn single(&self) -> Option<&T> {
         match self {
             ShortVec::Single(v) => Some(v),
             _ => None,
         }
     }
+
+    #[inline]
+    pub fn len(&self) -> usize {
+        match self {
+            ShortVec::Empty => 0,
+            ShortVec::Single(_) => 1,
+            ShortVec::Multi(ref v) => v.len(),
+        }
+    }
+
+    pub fn insert(&mut self, index: usize, elt: T) {
+        assert!(
+            index <= self.len(),
+            "insertion index (is {}) should be <= len (is {})",
+            index,
+            self.len()
+        );
+
+        *self = match core::mem::replace(self, ShortVec::Empty) {
+            ShortVec::Empty => ShortVec::Single(elt),
+            ShortVec::Single(item) => {
+                let mut items = vec![item];
```

Comment:
Nit: This might do two allocations, initialize the `Vec` with capacity 2.

Reply:
It won't - https://doc.rust-lang.org/src/alloc/macros.rs.html#50

Reply:
Not sure what that means. I'm expecting `vec!` to allocate space for exactly the number of elements, so the `insert` on the next line will trigger a resize.

Reply:
Oh, I haven't noticed that, that definitely is not cheap.
I'd recommend desugaring it into:
```rust
let items = if index == 0 {
    vec![elt, item]
} else {
    assert(index, 1);
    vec![item, elt]
};

Reply:
This assert is unnecessary as index<=1.